### PR TITLE
#182282682 - ADD PYTHON CODE STYLE CHECKS & LINT CHECKS PRE-COMMIT HOOKS

### PR DESCRIPTION
**Pull request by ChrisOnsando**
#### What does this Pull Request do?
- [ ] Adds the linters and code formatters for the project
#### Description of Task to be completed?
- [ ] Adds the linters and code formatters for the project
#### How should this be manually tested?
- [ ] running pre-commit run --all-files
#### Any background context you want to provide?
- [ ] Add python code style checks & lint checks  pre-commit hooks
#### What are the relevant pivotal tracker story?
- [ ] #5 https://www.pivotaltracker.com/n/projects/2570600/stories/182282682
#### Screenshots (if appropriate)
- [ ] N/A
#### Questions:
- [ ] None

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+- repo: https://github.com/pre-commit/mirrors-mypy
+  rev: 'v0.950'
+  hooks:
+    - id: mypy
+      exclude: "[a-zA-Z]*/(migrations)/(.)*"
+      additional_dependencies: [
+          # mypy dependencies
+          djangorestframework-stubs,
+          django-stubs,
+          python-dotenv,
```
Comment (mainamuriuki274) [inline .pre-commit-config.yaml:42]: are you using python-dotenv really?

File: Pipfile [hunk 2]
```diff
@@ -0,0 +1,15 @@
+[[source]]
+url = "https://pypi.python.org/simple"
+verify_ssl = true
+name = "pypi"
+
+[packages]
+dj-database-url = "*"
```
Comment (mainamuriuki274) [inline Pipfile:7]: Add package versions

File: mypy.ini [hunk 3]
```diff
@@ -0,0 +1,12 @@
+[mypy]
+plugins = mypy_django_plugin.main, mypy_drf_plugin.main
+ignore_missing_imports = True
+warn_unused_ignores = True
+warn_redundant_casts = True
+strict_optional = False
+
+[mypy.plugins.django-stubs]
+django_settings_module = "speaksfer.settings.local"
```
Comment (mainamuriuki274) [inline mypy.ini:9]: No need to specify .local

File: speaksfer/settings/base.py [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+# Quick-start development settings - unsuitable for production
+# See https://docs.djangoproject.com/en/4.0/howto/deployment/checklist/
+
+# SECURITY WARNING: keep the secret key used in production secret!
+SECRET_KEY = config("SECRET_KEY")
+
+# SECURITY WARNING: don't run with debug turned on in production!
+DEBUG = False
+
+ALLOWED_HOSTS = ["localhost"]
```
Comment (mainamuriuki274) [inline speaksfer/settings/base.py:28]: remove "localhost" , leave it as an empty list

File: speaksfer/settings/staging.py [hunk 5]
```diff
@@ -0,0 +1,5 @@
+from speaksfer.settings.base import ALLOWED_HOSTS
+
+DEBUG = True
+
+ALLOWED_HOSTS += ["localhost", "127.0.0.1"]
```
Comment (mainamuriuki274) [inline speaksfer/settings/staging.py:5]: we do not yet have a staging server so you can remove ALLOWED_HOSTS here

**Review by mainamuriuki274 — CHANGES_REQUESTED** [review]

File: Pipfile (hunk 2 above)
Comment (ChrisOnsando) [reply Pipfile:7]: Which django version do I add, 4.0 or 4.0.5?

File: mypy.ini (hunk 3 above)
Comment (ChrisOnsando) [reply mypy.ini:9]: Okay, I will be removing .local

File: .pre-commit-config.yaml (hunk 1 above)
Comment (mainamuriuki274) [reply .pre-commit-config.yaml:42]: you dont need dotenv cause you're using decouple

File: .pre-commit-config.yaml [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+          dj_database_url,
+          python-decouple,
+          psycopg2-binary,
+          # runtime dependencies
```
Comment (mainamuriuki274) [inline .pre-commit-config.yaml:46]: remove unnecessary comments

File: .flake8 [hunk 7]
```diff
@@ -0,0 +1,5 @@
+[flake8]
+    ignore = E203, E266, E501, W503, F403, F401
```
Comment (mainamuriuki274) [inline .flake8:2]: remove F403 and F401

**Review by mainamuriuki274 — CHANGES_REQUESTED** [review]

File: speaksfer/settings/base.py (hunk 4 above)
Comment (dalienst) [reply speaksfer/settings/base.py]: Okay. On it

File: speaksfer/settings/staging.py (hunk 5 above)
Comment (dalienst) [reply speaksfer/settings/staging.py]: Done

File: .flake8 (hunk 7 above)
Comment (dalienst) [reply .flake8:2]: Changes made

**Comment by mainamuriuki274** [comment]
Already done

**Pull request closed without merging**